### ISSUE TITLE
Mixed types iteration false positive

### ISSUE THREAD

**Issue by b1rdex** [kalessil/phpinspectionsea#242]
```php
class Test
{
    /**
     * @param array[] $b
     */
    protected function a(array $b)
    {
        $key = 'c';
        foreach ($b[$key] as $d) {
            assert(\is_array($d) === true);
        }
    }
}
```

`$b[$key]` is reported as containing `mixed` type.
But it's reported by PhpStorm `Ctrl+Q` as `array $b array|\array[]`.

`$d` type is correct and is `array`.

I'm not sure if it's phpinspectionsea bug or I should report it to jetbrains?

**Comment by b1rdex** [comment]
`$b[$key]` is not reported if I remove `array` from method signature (but not from phpdoc).

**Comment by kalessil** [comment]
It's our message. Weird, I'll check this on our side in first place.

**Comment by b1rdex** [comment]
I know this message is from plugin, but `array|array[]` types list is generated by phpstorm itself. I think it might be phpstorm bug that `array` from signature adds it to types list if `scalar[]` or `class[]` is already there.

**Comment by kalessil** [comment]
We have to fix this on our side, in fact fixed already. I pushed a fresh binary in case if you want to install fixed version.
